renderer/viewer/lib/workerProxy.ts

```
@@ -41,11 +53,30 @@ export const useWorkerProxy = <T extends { __workerProxy: Record<string, (...arg
+            || (typeof OffscreenCanvas !== 'undefined' && arg instanceof OffscreenCanvas)
+            || (typeof ImageData !== 'undefined' && arg instanceof ImageData)
+        }) : []
         worker.postMessage({
           type: prop,
+          msgId,
           args,
-        }, transfer as any[])
+        }, transfer)
+        return {
+          // eslint-disable-next-line unicorn/no-thenable
+          then (onfulfilled: (value: any) => void) {
```

Biome (1.9.4)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)